core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -118,6 +126,13 @@ private static final class CustomTypeSystem extends RelDataTypeSystemImpl {
     @Override public int getMaxNumericPrecision() {
       return 38;
     }
+
+    @Override public int getMaxPrecision(SqlTypeName typeName) {
+      if (typeName == SqlTypeName.TIMESTAMP) {
+        return SqlTypeName.MAX_DATETIME_PRECISION + 3; // Arbitrarily choose a different precision.
```

Comment:
Where is the +3 coming from?

Reply:
It's just a made-up number that differs from the default max.